sdk-python/src/agentrpc/http_client.py

```
@@ -0,0 +1,312 @@
+            "language": "python",
+            "version": self.headers.get("x-machine-sdk-version", "unknown"),
+        }
+        if functions:
+            payload["functions"] = functions
+        if tools:
+            payload["tools"] = tools
+
+        try:
+            return self.post("/machines", payload)
+        except Exception as e:
+            raise AgentRPCError(f"Failed to register machine: {str(e)}")
```

Ruff (0.13.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)